src/custom-element.js

```
@@ -117,96 +117,116 @@ export function stubElements(win) {
 
 
 /**
+ * Determines the layout of the element from the layout attribute, and if
+ * not specified, by considering hasNaturalDimensions() and the width,
+ * height, and sizes attributes.
+ * @param {!AmpElement} element
+ * @param {string} widthAttr
+ * @param {string} heightAttr
+ * @return {!Layout}
+ */
+function determineLayout(element, widthAttr, heightAttr) {
+  const layoutAttr = element.getAttribute('layout');
+  if (layoutAttr) {
+    const parsed = parseLayout(layoutAttr);
+    if (!parsed) {
+      throw new Error('Layout attr value invalid: ' + layoutAttr);
```

I believe `parseLayout` should throw this error.